src/simulation/hdf_simulation_store.jl

```diff
@@ -699,16 +699,23 @@ function _read_length(::Type{OptimizerStats}, store::HdfSimulationStore)
     return HDF5.read(HDF5.attributes(dataset), "columns")
 end
 
-function _write_dataset!(dataset, array, row_range)
-    if ndims(array) == 2
-        dataset[:, 1, row_range] = array
-    elseif ndims(array) == 3
-        dataset[:, :, row_range] = array
-        #elseif ndims(array) == 4
-        #    dataset[:, :, :, row_range] = array
-    else
-        error("ndims not supported: $(ndims(array))")
-    end
+function _write_dataset!(dataset, array::Array{Float64, 2}, row_range::UnitRange{Int64})
```

Comment:
Does row_range need to be strictly typed?

Reply:
make sure that it is continuous and Int so that can be used for indexing.

Reply:
I have found the following:
- Sometimes there is another type that can be passed and will work. The strict typing will cause an unnecessary failure. Not sure if that applies here.
- If you use duck typing here and the user passes the wrong type, Julia will fail with a meaningful error.
- Adding the type in this case doesn't help for dispatch or documentation.

Reply:
I am fine removing it. But Julia won't throw a meaningful error in this case if the user passes a non-contiguous vector. It will actually execute.